dace/transformation/passes/pattern_matching.py

```diff
@@ -538,7 +579,7 @@ def match_patterns(sdfg: SDFG,
         if len(singlestate_transformations) == 0:
             continue
         for state_id, state in enumerate(cfr.nodes()):
-            if states is not None and state not in states:
+            if (states is not None and state not in states) or not isinstance(state, SDFGState):
```

Comment:
change the order (the check `state not in states` may fail for some object types so it's better to start with that)